source/event_stream_rpc_server.c

```diff
@@ -0,0 +1,906 @@
+            aws_raise_error(AWS_ERROR_EVENT_STREAM_RPC_PROTOCOL_ERROR);
+            s_send_connection_level_error(
+                connection, AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PROTOCOL_ERROR, 0, &s_invalid_stream_id_error);
+            return;
+        }
+
+        /* INT32_MAX is the max stream id. */
+        if (stream_id > INT32_MAX) {
+            aws_raise_error(AWS_ERROR_EVENT_STREAM_RPC_PROTOCOL_ERROR);
+            s_send_connection_level_error(
+                connection, AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PROTOCOL_ERROR, 0, &s_invalid_stream_id_error);
+            return;
+        }
+
+        /* if the stream is is in the past, look it up from the continuation table. If it's not there, that's an error.
+         * if it is, find it and notify the user a message arrived */
+        if (stream_id <= connection->latest_stream_id) {
+            struct aws_hash_element *continuation_element = NULL;
+            if (aws_hash_table_find(&connection->continuation_table, &stream_id, &continuation_element) ||
+                !continuation_element) {
+                aws_raise_error(AWS_ERROR_EVENT_STREAM_RPC_PROTOCOL_ERROR);
+                s_send_connection_level_error(
+                    connection, AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PROTOCOL_ERROR, 0, &s_invalid_client_stream_id_error);
+                return;
+            }
+
+            continuation = continuation_element->value;
+            aws_event_stream_rpc_server_continuation_acquire(continuation);
+            continuation->continuation_fn(continuation, &message_args, continuation->user_data);
+            aws_event_stream_rpc_server_continuation_release(continuation);
+            /* now these are potentially new streams. Make sure they're in bounds, create a new continuation
+             * and notify the user the stream has been created, then send them the message. */
+        } else {
+            if (stream_id != connection->latest_stream_id + 1) {
+                aws_raise_error(AWS_ERROR_EVENT_STREAM_RPC_PROTOCOL_ERROR);
+                s_send_connection_level_error(
+                    connection,
+                    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PROTOCOL_ERROR,
+                    0,
+                    &s_invalid_new_client_stream_id_error);
+                return;
+            }
+
+            /* new streams must always have an operation name. */
+            if (operation_name.len == 0) {
+                aws_raise_error(AWS_ERROR_EVENT_STREAM_RPC_PROTOCOL_ERROR);
+                s_send_connection_level_error(
+                    connection, AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PROTOCOL_ERROR, 0, &s_missing_operation_name_error);
+                return;
+            }
+
+            continuation =
+                aws_mem_calloc(connection->allocator, 1, sizeof(struct aws_event_stream_rpc_server_continuation_token));
+            if (!continuation) {
+                aws_raise_error(AWS_ERROR_EVENT_STREAM_RPC_PROTOCOL_ERROR);
```

Comment:
Won't this overwrite the reason the alloc failed? Is that what you want here? It's not a protocol error.

Reply:
good catch. copy pasta error.